ios/Extensions.swift

```diff
@@ -42,32 +42,91 @@ extension UIView {
   }
   
   func animateQuickTap(
-    duration: TimeInterval = 0.15,
-    options: UIView.AnimationOptions = .curveEaseInOut,
+    duration: TimeInterval = 0.1,
     scale: CGFloat = 0.97
   ) {
-    UIView.animate(withDuration: duration, delay: 0, options: [options, .autoreverse], animations: {
+    let timingFunction = CAMediaTimingFunction(controlPoints: 0.25, 0.46, 0.45, 0.94)
+    
+    CATransaction.begin()
+    CATransaction.setAnimationTimingFunction(timingFunction)
+    
+    UIView.animate(withDuration: duration, delay: 0, options: [.autoreverse], animations: {
       self.transform = CGAffineTransform(scaleX: scale, y: scale)
     })
+    
+    CATransaction.commit()
   }
   
   func animateTapStart(
-    duration: TimeInterval = 0.15,
-    options: UIView.AnimationOptions = .curveEaseInOut,
+    duration: TimeInterval = 0.1,
     scale: CGFloat = 0.97,
     useHaptic: String? = nil
   ) {
     if useHaptic != nil {
       generateHapticFeedback(useHaptic!)
     }
-    UIView.animate(withDuration: duration, delay: 0, options: options, animations: {
+    
+    let timingFunction = CAMediaTimingFunction(controlPoints: 0.25, 0.46, 0.45, 0.94)
+    
+    CATransaction.begin()
+    CATransaction.setAnimationTimingFunction(timingFunction)
+    
+    UIView.animate(withDuration: duration) {
       self.transform = CGAffineTransform(scaleX: scale, y: scale)
-    })
+    }
+    
+    CATransaction.commit()
   }
   
-  func animateTapEnd(duration: TimeInterval = 0.15, options: UIView.AnimationOptions = .curveEaseInOut, scale: CGFloat = 0.97) {
-    UIView.animate(withDuration: duration, delay: 0, usingSpringWithDamping: 0.3, initialSpringVelocity: 10.0, options: options, animations: {
+  func animateTapEnd(duration: TimeInterval = 0.1, scale: CGFloat = 0.97) {
+    let timingFunction = CAMediaTimingFunction(controlPoints: 0.25, 0.46, 0.45, 0.94)
+    
+    CATransaction.begin()
+    CATransaction.setAnimationTimingFunction(timingFunction)
+    
+    UIView.animate(withDuration: duration) {
       self.transform = .identity
-    })
+    }
+    
+    CATransaction.commit()
+  }
+}
+
+extension UITableViewCell {
```

Comment:
Maybe I'm not that into Swift and I'm transferring my thinking from Java / JavaScript, but can't you rather focus on inheritance instead extensions? This mechanism overused makes code a bit dirty AFAIK. (Maybe I'm wrong)

Reply:
Every pattern applied in the wrong place might be overused. In this particular case, I agree with you, not every `UITableViewCell` should know how to generate a text image, especially a fixed-size preview. I believe this particular part of the code can be done using inheritance more efficiently, right as you say!